plugins/woocommerce/src/Gateways/PayPal/AddressRequirements.php

```diff
@@ -0,0 +1,348 @@
+		'CC',
+		'CO',
+		'KM',
+		'CD',
+		'ZR',
+		'CG',
+		'CK',
+		'CR',
+		'CI',
+		'HR',
+		'CU',
+		'CY',
+		'CZ',
+		'DK',
+		'DJ',
+		'DM',
+		'DO',
+		'TP',
+		'EC',
+		'EG',
+		'SV',
+		'GQ',
+		'ER',
+		'EE',
+		'ET',
+		'FK',
+		'FO',
+		'FM',
+		'FJ',
+		'FI',
+		'FR',
+		'GF',
+		'PF',
+		'TF',
+		'GA',
+		'GM',
+		'GE',
+		'DE',
+		'GH',
+		'GI',
+		'GR',
+		'GL',
+		'GD',
+		'GP',
+		'GU',
+		'GT',
+		'GG',
+		'GN',
+		'GW',
+		'GY',
+		'HT',
+		'HM',
+		'HN',
+		'HK',
+		'HU',
+		'IS',
+		'IN',
+		'ID',
+		'IR',
+		'IQ',
+		'IE',
+		'IM',
+		'IL',
+		'IT',
+		'JM',
+		'JP',
+		'JE',
+		'JO',
+		'KZ',
+		'KE',
+		'KI',
+		'KW',
+		'KG',
+		'LA',
+		'LV',
+		'LB',
+		'LS',
+		'LR',
+		'LY',
+		'LI',
+		'LT',
+		'LU',
+		'MO',
+		'MK',
+		'MG',
+		'MW',
+		'MY',
+		'MV',
+		'ML',
+		'MT',
+		'MH',
+		'MQ',
+		'MR',
+		'MU',
+		'YT',
+		'MX',
+		'MD',
+		'MC',
+		'MN',
+		'ME',
+		'MS',
+		'MA',
+		'MZ',
+		'MM',
+		'NA',
+		'NR',
+		'NP',
+		'NL',
+		'AN',
+		'NC',
+		'NZ',
+		'NI',
+		'NE',
+		'NG',
+		'NU',
+		'NF',
+		'MP',
+		'NO',
+		'OM',
+		'PK',
+		'PW',
+		'PS',
+		'PA',
+		'PG',
+		'PY',
+		'PE',
+		'PH',
+		'PN',
+		'PL',
+		'PT',
+		'PR',
+		'QA',
+		'RE',
+		'RO',
+		'RU',
+		'RW',
+		'WS',
+		'SM',
+		'ST',
+		'SA',
+		'SN',
+		'RS',
+		'CS',
+		'SC',
+		'SL',
+		'SG',
+		'SK',
+		'SI',
+		'SB',
+		'SO',
+		'ZA',
+		'GS',
+		'KR',
+		'ES',
+		'LK',
+		'SH',
+		'KN',
+		'LC',
+		'PM',
+		'VC',
+		'SD',
+		'SR',
+		'SJ',
+		'SZ',
+		'SE',
+		'CH',
+		'SY',
+		'TW',
+		'TJ',
+		'TZ',
+		'TH',
+		'TL',
+		'TG',
+		'TK',
+		'TO',
+		'TT',
+		'TN',
+		'TR',
+		'TM',
+		'TC',
+		'TV',
+		'UG',
+		'UA',
+		'AE',
+		'GB',
+		'US',
+		'UY',
+		'UM',
+		'UZ',
+		'VU',
+		'VA',
+		'VE',
+		'VN',
+		'VG',
+		'VI',
+		'WF',
+		'EH',
+		'YE',
+		'YU',
+		'ZM',
+		'ZW',
+	);
+
+	/**
+	 * Countries that require a postal code in the address.
+	 *
+	 * @var array
+	 */
+	private const COUNTRIES_REQUIRING_POSTAL_CODE = array(
+		'AR',
+		'AU',
+		'AT',
+		'BT',
+		'BR',
+		'CA',
+		'C2',
+		'CN',
+		'CC',
+		'KM',
+		'DK',
+		'FK',
+		'FO',
+		'FR',
+		'TF',
+		'GM',
+		'DE',
+		'GL',
+		'IT',
+		'JP',
+		'KI',
+		'KG',
+		'MR',
+		'YT',
+		'MX',
+		'NR',
+		'NL',
+		'NE',
+		'NU',
+		'NF',
+		'NO',
+		'PN',
+		'PL',
+		'SM',
+		'SG',
+		'ES',
+		'SH',
+		'PM',
+		'SR',
+		'SJ',
+		'SE',
+		'CH',
+		'TH',
+		'TK',
+		'TV',
+		'GB',
+		'US',
+		'UM',
+		'VA',
+		'WF',
+		'EH',
+	);
+
+	/**
+	 * Check if a country requires a city in the address.
+	 *
+	 * @param string $country_code The ISO 3166-1 alpha-2 country code.
+	 * @return bool
+	 */
+	public static function country_requires_city( $country_code ) {
+		return in_array( strtoupper( $country_code ), self::COUNTRIES_REQUIRING_CITY, true );
+	}
+
+	/**
+	 * Check if a country requires a postal code in the address.
+	 *
+	 * @param string $country_code The ISO 3166-1 alpha-2 country code.
+	 * @return bool
+	 */
+	public static function country_requires_postal_code( $country_code ) {
```

Comment:
Is it worth exposing these values as instance functions rather than static functions? That way it may be possible to mock or stub the methods in calling code, which is not possible at the moment. We can trigger the behaviour by picking country codes that do (or don't) match the criteria, but that makes tests dependent on the data definition (which is not necessarily a bad thing).

Reply:
And same as above RE requiring a string argument.
```suggestion
	public static function country_requires_postal_code( string $country_code ) {
```

Reply:
> Is it worth exposing these values as instance functions rather than static functions? That way it may be possible to mock or stub the methods in calling code, which is not possible at the moment. We can trigger the behaviour by picking country codes that do (or don't) match the criteria, but that makes tests dependent on the data definition (which is not necessarily a bad thing).

You mean using a singleton pattern like we have in Stripe? (so we can set a mocked instance) I will work on that

Reply:
Did it in https://github.com/woocommerce/woocommerce/pull/61706/commits/7cd7bcd7c82d04295e9bb5a4e5594a653876fb6a. This allows us to either set a mocked class version in the container